### PR TITLE
tests: Add unit tests for taxonomy and model family

### PR DESCRIPTION
**Pull request by hickeyma**
Add unit tests for taxonomy and model family.

### CODE REVIEW

File: tests/taxonomy.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+            rel_path (str): Relative path (from repository root) to the file.
+        Returns:
+            file_path: The path to the added file.
+        """
+        file_path = self.create_untracked(rel_path)
+        self._repo.index.add([rel_path])
+        self._repo.index.commit("new commit")
+        return file_path
+
+    def remove_file(self, rel_path: str) -> None:
```
Comment (markmc) [inline tests/taxonomy.py:82]: this is unused?

**Review by markmc — CHANGES_REQUESTED** [review]
Thanks @hickeyma just a couple of comments on the taxonomy test

File: tests/test_taxonomy.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class TestTaxonomy:
+    """Test taxonomy in instructlab.sdg.utils.taxonomy."""
+
+    @pytest.fixture(autouse=True)
+    def _init_taxonomy(self, taxonomy_dir):
+        self.taxonomy = taxonomy_dir
+
+    def test_read_taxonomy_leaf_nodes(self):
+        tracked_file = "compositional_skills/tracked/qna.yaml"
+        self.taxonomy.add_tracked(tracked_file)
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply tests/test_taxonomy.py]: Good idea, updated to pass contents.

File: tests/test_taxonomy.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        leaf_node = taxonomy.read_taxonomy_leaf_nodes(
+            self.taxonomy.root, TAXONOMY_BASE, TEST_CUSTOM_YAML_RULES
+        )
+        assert TEST_VALID_COMPOSITIONAL_SKILL_YAML not in leaf_node
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply tests/test_taxonomy.py]: yes, changed

File: tests/taxonomy.py (hunk 1 above)
Comment (hickeyma) [reply tests/taxonomy.py]: Good catch, removed.

File: tests/taxonomy.py [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        Returns:
+            file_path: The path to the created file.
+        """
+        taxonomy_path = Path(rel_path)
+        assert not taxonomy_path.is_absolute()
+        file_path = self.root.joinpath(taxonomy_path)
+        file_path.parent.mkdir(exist_ok=True, parents=True)
+        if not contents:
+            assert taxonomy_path.parts[0] == "compositional_skills"
+            file_path.write_text(TEST_VALID_COMPOSITIONAL_SKILL_YAML, encoding="utf-8")
```
[the remark this answers is not in the document]
Comment (hickeyma) [reply tests/taxonomy.py]: Sure, makes sense.

**Comment by hickeyma** [comment]
Thanks for review @markmc. Updated and ready for review again.

**Review by markmc — APPROVED** [review]
Thanks, @hickeyma !

**Pull request merged**